R/enum.R

```diff
@@ -17,19 +26,28 @@
 #' # With predefined values
 #' Symbol <- enum(c(Diamond = 1, Triangle = 2, Circle = 2))
 #' Symbol
+#'
 #' mySymbol <- Symbol$Diamond
 #' mySymbol
+#'
 #' @export
+
 enum <- function(enumValues) {
+  # if a vector, convert it to a list
   myEnum <- as.list(enumValues)
+
+  # save names of a vector or a list
   enumNames <- names(myEnum)
 
-  if (is.null(enumNames)) {
-    names(myEnum) <- myEnum
-  } else if (any(enumNames == "")) {
+  # check that none of the keys, if specified, are missing
+  if (!is.null(enumNames) && any(enumNames == "")) {
     stop(messages$errorEnumNotAllNames)
   }
 
+  # if no keys were specified, then use the values as keys themselves
+  names(myEnum) <- enumNames %||% myEnum
+
+  # return enumerated list
```

Comment:
This is what I meant with redundant comment. No need to remove it tho

Reply:
😅 

![image](https://user-images.githubusercontent.com/11330453/150769310-95b47558-a885-444d-93d5-241d01488d7f.png)

Reply:
lol